part-3/part-3.md

```diff
@@ -152,252 +153,289 @@ const source = from([1, 2, 3]).pipe(
+**Important!** Notice that **combineLatest** will only start emitting after each source `Observable` has emitted at least once. Previous incomplete values will be ignored 
 
 ```ts
-import { timer, combineLatest } from "rxjs";
+import { interval, combineLatest } from 'rxjs';
 
-// Первое значение вернет сразу, а остальные каждую секунду
 
-const sourceFirst = timer(0, 1000);
 
-// Первое значение через 0.5 секунды, а остальные каждую секунду
+const sourceFirst = interval(500);
 
-const sourceSecond = timer(500, 1000);
+
+const sourceSecond = interval(1000);
 
 const source = combineLatest([sourceFirst, sourceSecond]);
 
-// Выведет в консоль:
+// Will log:
 
-//. [0, 0] через 0.5 sec
+//. [1, 0] in 1 second
 
-//. [1, 0] через 1.0 sec
+//. [2, 0] in 1.5 seconds
 
-// [1, 1] через 1.5 sec
+// [3, 1] in 2 seconds
 
-// [2, 1] через 2.0 sec
+// and so on
 
-source.subscribe((value) => console.log(value));
+source.subscribe(value => console.log(value));
 ```
 
-Смотрите! Не смотря на то, что значения в потоках генерируются в разное время, нам удалось их свести к одному значению, что позволяет нам одновременно обрабатывать сразу два значения из потока.
+Despite the values of two streams are generated at different times, we managed to combine them into a single value, which allows us to handle values from two different streams simultaneously. **combineLatest** allows us to combine as many `Observables` as we want, not just two. 
 
-## **withLastestFrom()**
+## **withLatestFrom()**
 
-Как вы могли заметить, мы получали новое значение каждый раз, когда значение в одном из потоков менялось. А что если, нам надо получать новое значение только тогда, когда поменялся основной поток? Давайте это сделаем с помощью оператора **withLatestFrom()**.
+As you could see, we received a new value each time any of the two streams emitted. But what if we need to receive a value only when the source `Observable` emits? Let's achieve that using the **withLatestFrom()** operator.
 
-**withLatestFrom()** комбинирует текущий поток с переданным потоком, и создает новое значение только тогда, когда оно добавилось в основном потоке.
+**withLatestFrom()** combines the source `Observable` with another one, and emits a combined value only when the source one emits.
 
 ```ts
-import { timer } from "rxjs";
+import { interval } from 'rxjs';
 
-import { withLatestFrom } from "rxjs/operators";
+import { withLatestFrom } from 'rxjs/operators';
 
-// Первое значение вернет сразу, а остальные каждую секунду
+const sourceFirst = interval(1000);
 
-const sourceFirst = timer(0, 1000);
-
-// Первое значение через 0.5 секунды, а остальные каждую секунду
-
-const sourceSecond = timer(500, 1000);
+const sourceSecond = interval(500);
 
 const source = sourceFirst.pipe(withLatestFrom(sourceSecond));
```

Comment:
Nice to have: Examples with some additional log information. Something like this -

```
const source = sourceFirst.pipe(
  withLatestFrom(sourceSecond),
  map(([first, second]) => {
    return `First Source : ${first} Second Source : ${second}`;
  })
);
```

Reply:
Great suggestion, @prawin-s. This PR is mostly about translating the content from Russian to English. You're very welcome to add this suggestion to [the Part 3 discussion](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/3) or submit another PR 👐

Reply:
I agree. That's why I primarily wanted te examples to be reviewed

Reply:
I'll add a thread about code snippet review to [our Part 3 discussion](https://github.com/this-is-learning/rxjs-fundamentals-course/discussions/3) and link to @prawin-s' comment.